sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.controller.getInProgressApplicationDetails.e2e-spec.ts

```diff
@@ -0,0 +1,409 @@
+        parent2Info: SuccessWaitingStatus.Success,
+        outstandingAssessmentStatus: SuccessWaitingStatus.Success,
+      });
+  });
+
+  it("Should get application in-progress details when the student has partner as supporting user.", async () => {
+    // Arrange
+
+    const application = await saveFakeApplication(db.dataSource, { student });
+
+    // Create supporting users.
+    const partner = createFakeSupportingUser(
+      { application },
+      {
+        initialValues: {
+          supportingUserType: SupportingUserType.Partner,
+          supportingData: { totalIncome: 2000 },
+        },
+      },
+    );
+    await db.supportingUser.save([partner]);
+
+    // Create CRA income verifications for student and partner.
+    const studentCRAIncomeVerification = createFakeCRAIncomeVerification(
+      {
+        application,
+      },
+      { initialValues: { dateReceived: new Date() } },
+    );
+    const partnerCRAIncomeVerification = createFakeCRAIncomeVerification(
+      {
+        application,
+        supportingUser: partner,
+      },
+      { initialValues: { dateReceived: new Date() } },
+    );
+
+    await db.craIncomeVerification.save([
+      studentCRAIncomeVerification,
+      partnerCRAIncomeVerification,
+    ]);
+
+    // Mock assessment service to return the saved student assessment.
+    await mockOutstandingAssessment(appModule, application.currentAssessment);
+
+    const endpoint = `/students/application/${application.id}/in-progress`;
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({
+        id: application.id,
+        applicationStatus: application.applicationStatus,
+        pirStatus: application.pirStatus,
+        studentIncomeVerificationStatus: SuccessWaitingStatus.Success,
+        partnerIncomeVerificationStatus: SuccessWaitingStatus.Success,
+        partnerInfo: SuccessWaitingStatus.Success,
+        outstandingAssessmentStatus: SuccessWaitingStatus.Success,
+      });
+  });
+
+  it("Should get application in-progress details when there is an in-progress student assessment for an offering with later study start date.", async () => {
+    // Arrange
+
+    const currentApplication = await saveFakeApplication(db.dataSource, {
+      student,
+    });
+
+    // Create a previous application with an offering whose study start date and end date are much later
+    // than the current offering ones.
+    const futureApplication = await saveFakeApplication(db.dataSource, {
+      student,
+    });
+    const futureOffering = await db.educationProgramOffering.findOne({
+      select: {
+        id: true,
+        studyStartDate: true,
+        studyEndDate: true,
+      },
+      where: {
+        id: futureApplication.currentAssessment.offeringId,
+      },
+    });
```

Comment:
Is the future offering not returned like this??

`const futureOffering = futureApplication.currentAssessment.offering;`